new-lamassu-admin/src/pages/Analytics/graphs/TopMachinesBarGraph.js

```diff
@@ -0,0 +1,321 @@
+import BigNumber from 'bignumber.js'
+import * as d3 from 'd3'
+import * as R from 'ramda'
+import React, { memo, useCallback, useEffect, useMemo, useRef } from 'react'
+
+import {
+  java,
+  neon,
+  subheaderDarkColor,
+  fontColor,
+  fontSecondary
+} from 'src/styling/variables'
+
+const Graph = ({ data, period, timezone, machines, currency }) => {
+  const ref = useRef(null)
+
+  const AMOUNT_OF_MACHINES = 5
+  const BAR_PADDING = 0.15
+  const BAR_MARGIN = 10
+  const GRAPH_HEIGHT = 401
+  const GRAPH_WIDTH = 1163
+  const GRAPH_MARGIN = useMemo(
+    () => ({
+      top: 25,
+      right: 0.5,
+      bottom: 27,
+      left: 36.5
+    }),
+    []
+  )
+
+  const machinesClone = R.clone(machines)
+
+  // This ensures that the graph renders a minimum amount of machines
+  // and avoids having a single bar for cases with one machine
+  const filledMachines =
+    R.length(machines) >= AMOUNT_OF_MACHINES
+      ? machinesClone
+      : R.map(it => {
+          if (!R.isNil(machinesClone[it])) return machinesClone[it]
+          return { code: `ghostMachine${it}`, display: `` }
+        }, R.times(R.identity, AMOUNT_OF_MACHINES))
+
+  const txByDevice = R.reduce(
+    (acc, value) => {
+      acc[value.code] = R.filter(it => it.deviceId === value.code, data)
+      return acc
+    },
+    {},
+    filledMachines
+  )
+
+  const getDeviceVolume = deviceId =>
+    R.reduce(
+      (acc, value) => acc + BigNumber(value.fiat).toNumber(),
+      0,
+      txByDevice[deviceId]
+    )
+
+  const getDeviceVolumeByTxClass = deviceId =>
+    R.reduce(
+      (acc, value) => {
+        if (value.txClass === 'cashIn')
+          acc.cashIn += BigNumber(value.fiat).toNumber()
+        if (value.txClass === 'cashOut')
+          acc.cashOut += BigNumber(value.fiat).toNumber()
+        return acc
+      },
+      { cashIn: 0, cashOut: 0 },
+      txByDevice[deviceId]
+    )
+
+  const devicesByVolume = R.sort(
+    (a, b) => b[1] - a[1],
+    R.toPairs(
+      R.reduce(
+        (acc, value) => {
+          acc[value.code] = getDeviceVolume(value.code)
+          return acc
+        },
+        {},
+        filledMachines
+      )
+    )
+  )
+
+  const topMachines = R.take(AMOUNT_OF_MACHINES, devicesByVolume)
+
+  const txClassVolumeByDevice = R.reduce(
+    (acc, value) => {
+      acc[value[0]] = getDeviceVolumeByTxClass(value[0])
+      return acc
+    },
+    {},
+    topMachines
+  )
```

Comment:
Something like this?
```suggestion
  const txClassVolumeByDevice = R.fromPairs(
    R.map(v => [v[0], getDeviceVolumeByTxClass(v[0])], topMachines)
  )
```